pymarsys/contact.py

```diff
@@ -32,3 +32,290 @@ def create(self, contact, key_id=None, source_id=None):
+        :param key_values: Value of the keyId.
+        :param fields: Define which system fields to include in the output.
+        :return: Values of specified fields for contacts.
+        """
+        query_endpoint = '{}/{}/'.format(self.endpoint, 'getdata')
+        payload = {
+            'keyId': key_id,
+            'keyValues': key_values,
+        }
+
+        if fields:
+            payload['fields'] = fields
+
+        return self.connection.make_call(
+            'POST',
+            query_endpoint,
+            payload=payload
+        )
+
+    def get_history(self,
+                    contacts,
+                    start_date=None,
+                    end_date=None):
+        """
+        Returns a list of email campaign launch data for specified contacts,
+        can also be restricted to specified timeframe (optional).
+        :param contacts: Integer array which contains the contact IDs to
+        include.
+        :param start_date: yyyy-mm-dd formatted date string used to filter
+        emails by the date the launch was initiated.
+        :param end_date: yyyy-mm-dd formatted date string used to filter
+        emails by the date the launch completed.
+        :return: List of email campaign launch data.
+        """
+        query_endpoint = '{}/{}/'.format(self.endpoint, 'getcontacthistory')
+        payload = {
+            'contacts': contacts,
+        }
+
+        if start_date:
+            payload['startDate'] = start_date
+
+        if end_date:
+            payload['endate'] = end_date
+
+        return self.connection.make_call(
+            'POST',
+            query_endpoint,
+            payload=payload
+        )
+
+    def fetch_internal_id(self,
+                          key_field_id,
+                          key_field_value):
+        """
+        Returns the internal ID of a contact based on a specified key ID.
+        :param key_field_id: ID of the key field to use.
+        :param key_field_value: Key field value to use in the query.
+        :return: Internal ID of a contact.
+        """
+        params = {
+            key_field_id: key_field_value,
+        }
+
+        return self.connection.make_call(
+            'GET',
+            self.endpoint,
+            params=params
+        )
+
+    def check_ids(self,
+                  key_id,
+                  external_ids,
+                  get_multiple_ids=None):
+        """
+        Generates a list of existing contacts and errors indexed by a
+        specified key ID. Errors are collected if the key_id is invalid, if
+        no contact is found or if more than one contact is found with the
+        same key_id value.
+        :param key_id: Key which identifies the contacts
+        :param external_ids: Values specified in the key_id for those
+        contacts whose internal IDs the customer wants to receive.
+        :param get_multiple_ids: All contact IDs are listed in the case of
```

Comment:
change to get duplicated or something similar, it would make the parameter more explicit.